From 50ef3578dbde6023f7fb82f27288e857c033988b Mon Sep 17 00:00:00 2001
From: Josh Poimboeuf <jpoimboe@redhat.com>
Date: Thu, 4 Sep 2014 18:54:26 -0400
Subject: [kernel] trace: Remove unused function ftrace_off_permanent()

Message-id: <806ed7a10d0d0af131d1f4ad3355c5f31b6a9ffc.1409856554.git.jpoimboe@redhat.com>
Patchwork-id: 91248
O-Subject: [RHEL7.1 PATCH 07/23] tracing: Remove unused function ftrace_off_permanent()
Bugzilla: 1113829
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1113829

commit 042b10d83d05174e50ee861ee3aca55fd6204324
Author: Steven Rostedt (Red Hat) <rostedt@goodmis.org>
Date:   Wed Nov 6 12:19:37 2013 -0500

    tracing: Remove unused function ftrace_off_permanent()

    In the past, ftrace_off_permanent() was called if something
    strange was detected. But the ftrace_bug() now handles all the
    anomolies that can happen with ftrace (function tracing), and there
    are no uses of ftrace_off_permanent(). Get rid of it.

    Signed-off-by: Steven Rostedt <rostedt@goodmis.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/kernel.h b/include/linux/kernel.h
index 8f8582c..c12aa4e 100644
--- a/include/linux/kernel.h
+++ b/include/linux/kernel.h
@@ -511,7 +511,6 @@ void tracing_snapshot_alloc(void);
 
 extern void tracing_start(void);
 extern void tracing_stop(void);
-extern void ftrace_off_permanent(void);
 
 static inline __printf(1, 2)
 void ____trace_printk_check_format(const char *fmt, ...)
@@ -650,7 +649,6 @@ extern void ftrace_dump(enum ftrace_dump_mode oops_dump_mode);
 #else
 static inline void tracing_start(void) { }
 static inline void tracing_stop(void) { }
-static inline void ftrace_off_permanent(void) { }
 static inline void trace_dump_stack(void) { }
 
 static inline void tracing_on(void) { }
diff --git a/kernel/trace/trace.c b/kernel/trace/trace.c
index 618792e..cf12eaa 100644
--- a/kernel/trace/trace.c
+++ b/kernel/trace/trace.c
@@ -1247,21 +1247,6 @@ int is_tracing_stopped(void)
 }
 
 /**
- * ftrace_off_permanent - disable all ftrace code permanently
- *
- * This should only be called when a serious anomally has
- * been detected.  This will turn off the function tracing,
- * ring buffers, and other tracing utilites. It takes no
- * locks and can be called from any context.
- */
-void ftrace_off_permanent(void)
-{
- tracing_disabled = 1;
- ftrace_stop();
- tracing_off_permanent();
-}
-
-/**
  * tracing_start - quick start of the tracer
  *
  * If tracing is enabled but was stopped by tracing_stop,
-- 
1.7.1